File: Problemset/0027-Remove Element/Two Pointers.rb
# @param {Integer[]} nums
# @param {Integer} val
# @return {Integer}
def remove_element(nums, val)
    ret = 0

    for i in 0...nums.length
        if nums[i] != val
            nums[ret] = nums[i]
            ret += 1
        end
    end

    return ret
end

File: Problemset/0027-Remove Element/Two Pointers.rs
impl Solution {
    pub fn remove_element(nums: &mut Vec<i32>, val: i32) -> i32 {
        let mut ans = 0;
        for i in 0..nums.len() {
            if nums[i] != val {
                nums[ans] = nums[i];
                ans += 1;
            }
        }
        ans as i32
    }
}
